pkg/datasources/procedures.go

```diff
@@ -64,50 +64,52 @@ var proceduresSchema = map[string]*schema.Schema{
 
 func Procedures() *schema.Resource {
 	return &schema.Resource{
-		Read:   ReadProcedures,
-		Schema: proceduresSchema,
+		ReadContext: ReadContextProcedures,
+		Schema:      proceduresSchema,
 	}
 }
 
-func ReadProcedures(d *schema.ResourceData, meta interface{}) error {
+func ReadContextProcedures(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
 	databaseName := d.Get("database").(string)
 	schemaName := d.Get("schema").(string)
 
-	currentProcedures, err := snowflake.ListProcedures(databaseName, schemaName, db)
-	if errors.Is(err, sql.ErrNoRows) {
-		// If not found, mark resource to be removed from state file during apply or refresh
-		log.Printf("[DEBUG] procedures in schema (%s) not found", d.Id())
-		d.SetId("")
-		return nil
-	} else if err != nil {
-		log.Printf("[DEBUG] unable to parse procedures in schema (%s)", d.Id())
-		d.SetId("")
-		return nil
+	req := sdk.NewShowProcedureRequest()
+	if databaseName != "" {
+		req.WithIn(&sdk.In{Database: sdk.NewAccountObjectIdentifier(databaseName)})
+	}
+	if schemaName != "" {
+		req.WithIn(&sdk.In{Schema: sdk.NewDatabaseObjectIdentifier(databaseName, schemaName)})
+	}
+	procedures, err := client.Procedures.Show(ctx, req)
+	if err != nil {
+		return diag.FromErr(err)
```

Comment:
returning a warning now, and setting id to "". but I think it actually doesn't matter for procedures, as there are some built in functions which will always get returned. also for data source we probably do want to return empty list, what if someone wanted to do a for loop on a empty list? that would be fine, but in your case it wouldn't be fine since the resource doesn't exist

Reply:
I guess that's something we should discuss further, added to our list and we can move on with what we have right now